### ISSUE TITLE
Issue with installing/using latest version of org-mode

### ISSUE THREAD

**Issue by fast-90** [radian-software/straight.el#988]
I am having issues with using the latest version of org-mode. I have the following in my config:

```elisp
(use-package org
  :straight t)
```

Most things seem to work, `org-version` shows the expected version and path. However, when I try to show an org-agenda view (any view) I get the following error:
`Symbol's value as variable is void: org-link--description-folding-spec`

I have also tried to use the emacs-straight/org-mode repo, and have also tried to delete my eln-cache folder but it didn't solve my issue. I have the exact same issue on my Macbook as well as on my Windows/WSL laptop.

Using `:straight (:type built-in)` works, but I would like to use the latest version of org-mode.

Not sure if this is an issue with straight or org-mode, but I haven't found any references on line to `org-link--description-folding-spec` and I know that other people have had issues with straight in combination with org-mode before, hence I am trying my luck here. 

Any help would be appreciated!

**Comment by raxod502** [comment]
Latest version of Org seems to be working for me (3303a54d74805cdd2b6eda37285f259949f7ba2e). The variable in question seems to be defined here, maybe you can check if your copy differs? https://git.savannah.gnu.org/cgit/emacs/org-mode.git/tree/lisp/ol.el?h=3303a54d74805cdd2b6eda37285f259949f7ba2e#n631

**Comment by fast-90** [comment]
@raxod502 Thanks for the pointer. After some experimentation, it appeared that the issue above was caused by the fact that I am using a literate config in org (where I call the latest version of org), which is called by `org-babel-load-file` in init.el using the built-in version (hence the mix up of versions).

I have "solved" this issue now by moving all my straight, use-package and org configs to init.el instead of my literate config.org, however I know face a new issue where I get "Invalid function: org-encode-time" error when trying to run a `org-clock-report`. Note that it initially works fine, however at some random point it stops working. Not sure if I explained it well, so [here is a short video](https://youtu.be/wSrqDHNlMNA) to show what is happening (this is using a minimal config with just the necessary code for straight, use-package and org in init.el, and I am continuously pressing C-c to refresh the org clock table). Manually invoking `(org-matcher-time "<2022-08-15 Mon 00:00>")` immediately gives me the same issue btw (in case someone wants to replicate this issue faster than continuously pressing C-c on an `org-clock-report`).

If I manually delete the contents of the eln-cache folder and restart emacs, I can use `org-clock-report` again, but at some point it will just give me the same error just like in the video. 

If I look in my user-emacs-directory/straight/repos/org/lisp/ob-macs.el file I see the code defining the org-encode-time macro, but not sure why it is not working for me.

Current version of Emacs: GNU Emacs 29.0.50 (build 1, x86_64-apple-darwin21.5.0, NS appkit-2113.50 Version 12.4 (Build 21F79)) of 2022-07-10.

Any thoughts on this new issue?
